airrohr-firmware/airrohr-firmware.ino

```diff
@@ -4240,9 +4244,13 @@ static bool acquireNetworkTime() {
 	retryCount = 0;
 	configTime(0, 0, WiFi.gatewayIP().toString().c_str());
 	while (retryCount++ < 20) {
-		// later than 2000/01/01:00:00:00
+		now = time(nullptr);
+#if defined(ESP8266)
 		if (sntp_time_is_set) {
-			now = time(nullptr);
+#endif
+#if defined(ESP32)
+		if (now > 17897*24*60*60) { // later than 2019/01/01:00:00:00
```

Comment:
doesn't that mean that sntp_time_is_set should not be true?

Reply:
I'm not sure if I understood your questions correctly.
sntp_time_is_set is initialized with false and only set in the callback time_is_set().
When the callback is called in the first try with "ntp.org", the second part (the change from line 4247 to 4252) will never be reached because the function returns.

Reply:
Right, I agree with your assessment, sorry, didn't read the code closely. I wonder if we can remove all of the ESP conditionals in the code and use both code paths the same way?

Reply:
Yes, in this specific case the ESP32 path will work for both. The error checking of the ESP8266 is more strict. The fallback in the ESP32 case to use the local router will not be triggered if the RTC has a date after 2019-01-01 and this will happen on a reboot.

That means:

- when ntp.org cannot be reached on the first boot it will fallback to the local router
- on the second boot the RTC will already have a valid date and don't do a fallback to the local router

To solve that, I would remove the fallback and replace it by the following solution:
Set multiple servers with one call:
1. ntp.org
2. the ntp server(s) announced by dhcp
3. the local router (actually this shall not be necessary because in many cases this is already covered by 2.)

roughly the code would be the following, but I need to check how to mix dhcp and fixed servers:
```
sntp_servermode_dhcp(1);
sntp_setservername(0, "pool.ntp.org");
sntp_setserver(1, WiFi.gatewayIP());

```

Reply:
I would try local first before going to the internet.. I implemented router/pool in my pr that I just opened.

Searching for ntp information in dhcp does sound like a good idea though.